packages/components/src/components/hds/table/index.ts

```diff
@@ -4,29 +4,110 @@
  */
 
 import Component from '@glimmer/component';
-import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
 import { assert } from '@ember/debug';
+import { tracked } from '@glimmer/tracking';
+import type { ComponentLike } from '@glint/template';
+
+import type { HdsFormCheckboxBaseSignature } from '../form/checkbox/base';
+import type { HdsTableTdArgs } from './td.ts';
+import type { HdsTableThArgs } from './th.ts';
+import type { HdsTableThSortArgs } from './th-sort.ts';
+import type { HdsTableTrArgs } from './tr.ts';
+import {
+  HdsTableDensityValues,
+  HdsTableThSortOrderValues,
+  HdsTableVerticalAlignmentValues,
+} from './types.ts';
+import type {
+  HdsTableColumn,
+  HdsTableDensities,
+  HdsTableHorizontalAlignment,
+  HdsTableSelectableRow,
+  HdsTableSortingFunction,
+  HdsTableThSortOrder,
+  HdsTableVerticalAlignment,
+} from './types';
+
+export const DENSITIES: string[] = Object.values(HdsTableDensityValues);
+export const DEFAULT_DENSITY = HdsTableDensityValues.Medium;
+
+export const VALIGNMENTS: string[] = Object.values(
+  HdsTableVerticalAlignmentValues
+);
+export const DEFAULT_VALIGN = HdsTableVerticalAlignmentValues.Top;
+
+export interface HdsTableArgs {
+  Args: {
+    align?: HdsTableHorizontalAlignment;
+    caption?: string;
+    columns?: HdsTableColumn[];
+    density?: HdsTableDensities;
+    identityKey?: string;
+    isFixedLayout?: boolean;
+    isSelectable?: boolean;
+    isStriped?: boolean;
+    model: Record<string, unknown>;
+    onSelectionChange?: (selection: {
+      selectionKey: string;
+      selectionCheckboxElement:
+        | HdsFormCheckboxBaseSignature['Element']
+        | undefined;
+      selectedRowsKeys: string[];
+      selectableRowsStates: {
+        selectionKey: string;
+        isSelected: boolean | undefined;
+      }[];
+    }) => void;
+    onSort?: (sortBy: string, sortOrder: HdsTableThSortOrder) => void;
+    selectionAriaLabelSuffix?: string;
+    sortBy?: string;
+    sortedMessageText?: string;
+    sortOrder?: HdsTableThSortOrder;
+    valign?: HdsTableVerticalAlignment;
+  };
+  Blocks: {
+    head?: [
+      {
+        Tr: ComponentLike<HdsTableTrArgs>;
+        Th: ComponentLike<HdsTableThArgs>;
+        ThSort: ComponentLike<HdsTableThSortArgs>;
+        sortBy: string | undefined;
+        sortOrder: HdsTableThSortOrder;
+        setSortBy: (column: string) => void;
+      },
+    ];
+    body?: [
+      {
+        Td: ComponentLike<HdsTableTdArgs>;
+        Tr: ComponentLike<HdsTableTrArgs>;
+        Th: ComponentLike<HdsTableThArgs>;
+        ThSort?: ComponentLike<HdsTableThSortArgs>;
+        data?: string;
```

Comment:
Should this be a string, or something like `unknown`? It should be the same type as `@model` members I think.

Reply:
It is actually the identityKey which is a string
{{#each (sort-by this.getSortCriteria @model) key=this.identityKey as |record|}}

Reply:
Gonna dig deeper into the ember/syntax and see. Good call out.